pkg/build/controller/controller.go

```diff
@@ -326,7 +323,8 @@ func (bc *BuildPodController) HandlePod(pod *kapi.Pod) error {
 
 	// Update the build object when it progress to a next state or the reason for
 	// the current state changed.
-	if build.Status.Phase != nextStatus && !buildutil.IsBuildComplete(build) {
+	if (!hasBuildPodNameAnnotation(build) || build.Status.Phase != nextStatus) && !buildutil.IsBuildComplete(build) {
```

Comment:
hmmm.. any idea why we were/are excluding complete builds here?  note that line 337/335 actually then attempts to set a timestamp if the build is complete, which will never get invoked here....